### PR TITLE
Fix on UX and recipe loop

### PR DESCRIPTION
**Pull request by alexbourret**

### CODE REVIEW

File: custom-recipes/youtube-items/recipe.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
         "iconColor": "#FF0000"
     },
     "kind": "PYTHON",
+    "selectableFromDataset": "input_datasets_name",
```
Comment (tdesfont) [inline custom-recipes/youtube-items/recipe.json:9]: Good point.

File: custom-recipes/youtube-items/recipe.py [hunk 2]
```diff
@@ -32,16 +32,11 @@
     id = row[id_column_name]
```
Comment (tdesfont) [inline custom-recipes/youtube-items/recipe.py:32]: Would you want to get rid of variable id ?

File: parameter-sets/oauth-credentials/parameter-set.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
         "description": "",
         "icon": "icon-youtube-play"
     },
-    "defaultDefinableInline": true,
+    "defaultDefinableInline": false,
```
Comment (tdesfont) [inline parameter-sets/oauth-credentials/parameter-set.json:7]: What does this do ?

File: python-lib/youtube_client.py [hunk 4]
```diff
@@ -274,7 +276,7 @@ def get_error(self, response):
     def format_template(self, template, **kwargs):
         try:
             template = template.format(**kwargs)
-        except KeyError as key:  # This has to go
+        except KeyError:
             template = ""
```
Comment (tdesfont) [inline python-lib/youtube_client.py:280]: You might be willing to log this if no sensitive information is displayed ?

File: python-lib/youtube_client.py [hunk 5]
```diff
@@ -340,12 +342,21 @@ def store_next_page(self, url, headers, params, json_response):
...
 
+    def has_data_to_process(self):
+        if self.initial_data_process_loop:
+            self.initial_data_process_loop = False
+            return True
+        else:
+            return self.has_next_page()
+
     def has_next_page(self):
         return self.next_page is None or self.next_page != {}
```
Comment (tdesfont) [inline python-lib/youtube_client.py:353]: **!!!**
Should it not be 
`return self.next_page is not None and self.next_page != {}` ?

**Review by tdesfont — CHANGES_REQUESTED** [review]

File: parameter-sets/oauth-credentials/parameter-set.json (hunk 3 above)
Comment (alexbourret) [reply parameter-sets/oauth-credentials/parameter-set.json:7]: It makes the default inline option off for the oauth-credential, which in turn should makes unavailable the "Manually defined" option in the user's preset. It make no sense to define the oauth credentials at the datasets / recipes level so overall it's less confusing for the user.

**Review by tdesfont — APPROVED** [review]

**Pull request merged**